### PR TITLE
Cleanup "Upgrading Version 8" page of User Manual page for Gradle 8.9RC1 release

### PR DESCRIPTION
**Pull request by lkasso**
### Details
This is a Documentation change ONLY.

### Context 
This is an update for the User Manual's [Upgrading Version 8](https://github.com/gradle/gradle/blob/3c9b45ac28bedbbbac4751f3d497f4dd9a34aa91/platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc) page for Gradle 8.9 RC1. **--> [BUILD LINK FOR REVIEW](https://builds.gradle.org/repository/download/Gradle_Release_Check_BuildDistributions/83552575:id/distributions/gradle-8.9-docs.zip!/gradle-8.9-20240612234831%2B0000/docs/userguide/upgrading_version_8.html) <--**

### CODE REVIEW

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle test BD

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 1]
```diff
@@ -41,40 +41,39 @@ The previous step will help you identify potential problems by issuing deprecati
...
+
+In previous versions of Gradle, toolchain provisioning could leave a partially provisioned toolchain in place **with a marker file indicating that the toolchain was fully provisioned**. This could lead to strange behavior with the toolchain. In Gradle 8.9, the toolchain is fully provisioned before the marker file is written. However, to not detect potentially broken toolchains, a different marker file (`.ready`) is used. This means all your existing toolchains will be re-provisioned the first time you use them with Gradle 8.9. Gradle 8.9 also writes the old marker file (`provisioned.ok`) to indicate that the toolchain was fully provisioned. This means that if you return to an older version of Gradle, an 8.9-provisioned toolchain will **not** be re-provisioned.
```
Comment (ljacomet) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:45]: 💅 One sentence per line

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 2]
```diff
@@ -41,40 +41,39 @@ The previous step will help you identify potential problems by issuing deprecati
...
 
 ==== Compiling against Gradle implementation classpath
 
-In previous versions of Gradle, it was possible for Java projects that had no declared dependencies to implicitly compile against Gradle's runtime classes.
-This means it was possible for some projects to compile without any declared dependencies even though they referenced Gradle runtime classes.
-We believe this situation is very unlikely to occur in real projects, as IDE integration and test execution would not work correctly.
-If you need to use the Gradle API, declare a `gradleApi` dependency or use the `java-gradle-plugin` plugin.
+In previous versions of Gradle, Java projects that had no declared dependencies could implicitly compile against Gradle's runtime classes.
+This means that some projects were able to compile without any declared dependencies even though they referenced Gradle runtime classes.
+This situation is unlikely to arise in projects since IDE integration and test execution would be compromised. However, if you need to utilize the Gradle API, declare a `gradleApi` dependency or apply the `java-gradle-plugin` plugin.
```
Comment (ljacomet) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:61]: 💅 One sentence per line

**Review by ljacomet — APPROVED** [review]
LGTM

Only minor comments on keeping one sentence per line. Did not repeat it for all occurences.

**Comment by lkasso** [comment]
Moved to https://github.com/gradle/gradle/pull/29542

**Pull request closed without merging**